modules/gdscript/gdscript_editor.cpp

```diff
@@ -2651,14 +2654,9 @@ ::Error GDScriptLanguage::complete_code(const String &p_code, const String &p_pa
 					String opt = E.strip_edges();
 					if (opt.is_quoted()) {
 						r_forced = true;
-						String idopt = opt.unquote();
-						if (idopt.replace("/", "_").is_valid_identifier()) {
-							ScriptCodeCompletionOption option(idopt, ScriptCodeCompletionOption::KIND_NODE_PATH);
-							options.insert(option.display, option);
```

Comment:
This needs review/testing as I changed the logic a bit.

Reading this code, it seemed to me that we might be passing `idopt` as unquoted by oversight. But if I quote it back, then I end up with the exact same code as the `else` branch, so the check is useless.

Now, maybe `idopt.replace("/", "_")` was mistakenly assumed to mutate `idopt` (it doesn't).

It could be that the actual correct code would be to pass `idopt.replace("/", "_").quote(quote_style)` instead in the `option`.

Reply:
From some quick testing it seems to work ok with this PR to complete stuff like `get_node('Button1/Button2')`.

But I'm not sure what the `replace("/", "_")` meant to check exactly in the context of `get_node()`.

Reply:
I believe this is the code for completing the `$` notation, so if it's a valid identifier it doesn't need quotes. The `replace()` is there to check the whole path at once, since something like `Parent/Child` is okay to use without quotes.

I'm just not sure what the outer `if (opt.is_quoted())` is for, since this seems to always come quoted and there's not branching handling the case if they aren't.

Reply:
Thanks, I fixed the issue by re-adding the special case and refactored the code to solve the `is_quoted()` issue (I still handle it just in case as other an `unquote()` would be destructive).

I added comments to make things clear.

Reply:
> I still have an issue that triggering auto-completion when quotes are added creates a bug where two quotes are added at the end, e.g.:

Confirmed that this bug happens in `master` too.